### PR TITLE
Add Panel widget

### PR DESCRIPTION
**Pull request by shkolin**
| Q             | A
| ------------- | ---
| Is bugfix?    | ❌
| New feature?  | ✔️
| Breaks BC?    | ❌
| Fixed issues  | -

### CODE REVIEW

File: docs/panel.md [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+
+$assetManager->register([
+    BulmaAsset::class,
+    BulmaJsAsset::class,
+]);
+
+$this->setCssFiles($assetManager->getCssFiles());
+$this->setJsFiles($assetManager->getJsFiles());
+
+$template =<<<HTML
```
Comment (samdark) [inline docs/panel.md:29]: Is it necessary to provide template every time when using the widget? If so, that's very inconvenient and I'd better do it with simple PHP and HTML...

File: src/Panel.php [hunk 2]
```diff
@@ -0,0 +1,306 @@
...
+    /**
+     * Set progress bar color.
+     *
+     * @var string $value Color class.
+     *
+     * @return self
+     */
+    public function color(string $value): self
+    {
+        if (!in_array($value, self::COLOR_ALL)) {
```
Comment (samdark) [inline src/Panel.php:133]:
```suggestion
        if (!in_array($value, self::COLOR_ALL, true)) {
```

File: src/Panel.php [hunk 3]
```diff
@@ -0,0 +1,306 @@
...
+        $linkOptions = ArrayHelper::getValue($item, 'linkOptions', []);
+        $items = ArrayHelper::getValue($item, 'items', []);
+        $itemsContainerOptions = ArrayHelper::getValue($item, 'itemsContainerOptions', []);
+
+        if ($url !== '') {
+            $linkOptions['href'] = $url;
+        }
+
+        if ($label === '') {
+            throw new InvalidArgumentException("The 'label' option is required.");
```
Comment (samdark) [inline src/Panel.php:229]:
```suggestion
            throw new InvalidArgumentException('The "label" option is required.');
```

File: src/Panel.php [hunk 4]
```diff
@@ -0,0 +1,306 @@
...
+        if ($this->isActive($item)) {
+            Html::addCssClass($linkOptions, 'is-active');
+        }
+
+        if (is_array($items) && !empty($items)) {
+            $linkOptions['href'] ??= '#' . $id;
+            $itemsContainerOptions['id'] ??= $id;
+
+            $itemsContainer = Html::beginTag('div', $itemsContainerOptions) . "\n";
+            foreach ($items as $item) {
```
Comment (samdark) [inline src/Panel.php:245]: `$item` overrides method argument.

File: src/Panel.php [hunk 5]
```diff
@@ -0,0 +1,306 @@
...
+
+    private function renderItem(array $item): string
+    {
+        $options = ArrayHelper::getValue($item, 'options', []);
+        $label = ArrayHelper::getValue($item, 'label', '');
+        $icon = ArrayHelper::getValue($item, 'icon', '');
+        $encode = ArrayHelper::getValue($item, 'encode', $this->encodeLabels);
+
+        if ($label === '') {
+            throw new InvalidArgumentException("The 'label' option is required.");
```
Comment (samdark) [inline src/Panel.php:264]:
```suggestion
            throw new InvalidArgumentException('The "label" option is required.');
```

**Review by samdark — CHANGES_REQUESTED** [review]

File: docs/panel.md (hunk 1 above)
Comment (shkolin) [reply docs/panel.md:29]: > Is it necessary to provide template every time when using the widget?

No, there is a default template https://github.com/yiisoft/yii-bulma/blob/c9e5db7240b5518a5de973a94bf83dee8f7cbd9f/src/Panel.php#L35

File: src/Panel.php (hunk 4 above)
Comment (shkolin) [reply src/Panel.php]: Fixed https://github.com/yiisoft/yii-bulma/pull/29/commits/214ffc7cecb6addbf21d8142c87b82d330b76d83

File: docs/panel.md (hunk 1 above)
Comment (samdark) [reply docs/panel.md:29]: Would it generate proper bulma markup?

Comment (shkolin) [reply docs/panel.md:29]: Yes. https://github.com/yiisoft/yii-bulma/blob/214ffc7cecb6addbf21d8142c87b82d330b76d83/tests/PanelTest.php#L188

**Review by samdark — APPROVED** [review]

**Comment by samdark** [comment]
Thanks.

**Pull request merged**